src/neo/SmartContract/Native/Oracle/OracleRequest.cs

```diff
@@ -0,0 +1,41 @@
+using Neo.IO;
+using Neo.VM;
+using Neo.VM.Types;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native.Oracle
+{
+    public class OracleRequest : IInteroperable
+    {
+        public UInt256 Txid;
+        public string Url;
+        public string Filter;
+        public UInt160 CallbackContract;
+        public string CallbackMethod;
+        public byte[] UserData;
+
+        public void FromStackItem(StackItem stackItem)
+        {
+            Array array = (Array)stackItem;
+            Txid = new UInt256(array[0].GetSpan());
+            Url = array[1].GetString();
+            Filter = array[2].GetString();
+            CallbackContract = new UInt160(array[3].GetSpan());
+            CallbackMethod = array[4].GetString();
+            UserData = array[5].GetSpan().ToArray();
```

Comment:
Preserve the `StackItem[]` here?

Reply:
Then you are unable to send the `ReferenceCounter` when deserializing.